Timestamp log file lines

mjuric opened the pull request:
Make log files more informative when debugging performance issues.

NOTE: I'm not sure if buildbot's scripts parse the log files in some way (I don't think they do) -- if they do, this change may break them.

mjuric committed:
_build.sh: Emit a message when setting up environment

mjuric committed:
Prepend UTC timestamps to lines in _build.log

ktlim commented on python/lsst/ci/build.py:
```diff
@@ -188,6 +190,7 @@ def _build_product(self, product, progress):
             # execute the build file from the product directory, capturing the output and return code
             process = subprocess.Popen(buildscript, stdout=subprocess.PIPE, stderr=subprocess.PIPE, cwd=productdir)
             for line in iter(process.stdout.readline, ''):
+                line = "[%s] %s" % (datetime.datetime.utcnow().isoformat(), line)
```
Must have a timezone indicator (at least "Z") here.  If we're keeping this for much longer, I think I'd want to consider using logging instead of custom output.

mjuric replied:
Agreed. Fixed & pushed.

mjuric committed:
bugfix: preserve EUPS_PATH on 'lsst-build build' invocation

Before this fix, the setups.sh script invoked from _build.sh would prepend
the path to the default stack to whatever was in $EUPS_PATH (and thus all
built packages would be installed into the default stack, irrespective of
what $EUPS_PATH was pointing to).

mjuric commented:
Also added a bugfix for environmental EUPS_PATH being forgotten in builds.

mjuric commented:
@ktlim @jhoblitt Take a look and let me know if this is OK to merge.

ktlim commented:
Looks OK to me.

jhoblitt commented:
Looks OK to me as well.  What is the use case for having `EUPS_PATH` set to a value other than what is set by `eups/current/bin/setups.sh`?

mjuric commented:
@jhoblitt It's about general consistency (i.e., if you set EUPS_PATH=.... in the environment, the expectation is that is the path that EUPS is seeing).

Also very helpful when working on EUPS code -- one wants to be able to set EUPS_PATH=... to some throwaway directory when trying things out.

PaulPrice commented:
I believe it can also contain multiple paths (colon-delimited) if you're using multiple repos.
